MetaMask/metamask-extension
# Adopt ApprovalType from core

## Explanation


Adopt `ApprovalType` to split the concept of `MESSAGE_TYPE` and approval requests/confirmations leaving room for a more cohesive usage between extension and mobile.

resolves: https://github.com/MetaMask/MetaMask-planning/issues/424

## Manual Testing Steps

- Go to the [test dapp](https://metamask.github.io/test-dapp/)
- Try Eth sign, personal sign and Sign typed data.
- Transfer eth and execute Ethereum Chain Interactions

All should maintain the same behaviour as changes were only replacing the MESSAGE_TYPE to ApprovalType from code for approvals/confirmations without any change in the values.



## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
